pkg/android/aab.go

```diff
@@ -0,0 +1,87 @@
+package android
+
+import (
+	"fmt"
+	"github.com/bugsnag/bugsnag-cli/pkg/log"
+	"github.com/bugsnag/bugsnag-cli/pkg/utils"
+	"path/filepath"
+)
+
+func MergeUploadOptionsFromAabManifest(path string, apiKey string, applicationId string, buildUuid string, versionCode string, versionName string) (map[string]string, error) {
+
+	var manifestData map[string]string
+	var err error
+	var aabManifestPath string
+	aabUploadOptions := make(map[string]string)
+
+	aabUploadOptions["apiKey"] = apiKey
+	aabUploadOptions["applicationId"] = applicationId
+	aabUploadOptions["buildUuid"] = buildUuid
+	aabUploadOptions["versionCode"] = versionCode
+	aabUploadOptions["versionName"] = versionName
+
+	if apiKey == "" || applicationId == "" || buildUuid == "" || versionCode == "" || versionName == "" {
+
+		aabManifestPathExpected := filepath.Join(path, "base", "manifest", "AndroidManifest.xml")
+
+		if utils.FileExists(aabManifestPathExpected) {
+			aabManifestPath = aabManifestPathExpected
+		} else {
+			return nil, fmt.Errorf("AndroidManifest.xml not found in AAB file")
+		}
+
+		log.Info("Reading data from AndroidManifest.xml")
+
+		manifestData, err = ReadAabManifest(filepath.Join(aabManifestPath))
+
+		if err != nil {
+			return nil, fmt.Errorf("unable to read data from " + path + " " + err.Error())
+		}
+
+		if aabUploadOptions["apiKey"] == "" {
+			aabUploadOptions["apiKey"] = manifestData["apiKey"]
+			if aabUploadOptions["apiKey"] != "" {
+				log.Info("Using " + aabUploadOptions["apiKey"] + " as API key from AndroidManifest.xml")
+			}
+		}
```

Comment:
This is probably a little more succinct?

```suggestion
		if aabUploadOptions["apiKey"] == "" && manifestData["apiKey"] != "" {
			aabUploadOptions["apiKey"] = manifestData["apiKey"]
			log.Info("Using " + manifestData["apiKey"] + " as API key from AndroidManifest.xml")
		}
```